current/link-analysis.md

```diff
@@ -62,7 +62,17 @@ links.saveAsTextFile("links-all-apple/")
 
 ### Scala DF
 
-TODO
+```scala
+import io.archivesunleashed._
+import io.archivesunleashed.df._
+
+RecordLoader.loadArchives("example.arc.gz", sc).webgraph()
+  .groupBy(RemovePrefixWWWDF(ExtractDomainDF($"src")).as("src"), RemovePrefixWWWDF(ExtractDomainDF($"dest")).as("dest"))
+  .count()
+  .filter(($"src".isNotNull) || ($"dest".isNotNull))
```

Comment:
I would prefer above since it makes more sense logically.
Don't need the paren, right?
And unless `filter` works differently in DF vs. RDD, should it be `&&` instead of `||`?

Reply:
That is, we only want to keep all links where _both_ the src and dest are *not* null?

Reply:
Actually, now that I think about it, wouldn't it make more sense to push
`.filter($"src".isNotNull && $"dest".isNotNull)` into `.webgraph()` itself?

I can't imagine the user wanting nulls in the webgraph?